pkg/plugins/gateway/gateway.go

```diff
@@ -421,6 +428,24 @@ func (s *Server) HandleResponseHeaders(ctx context.Context, requestID string, re
 		})
 	}
 
+	var isProcessingError bool
+	var processingErrorCode int
+	for _, headerValue := range b.ResponseHeaders.Headers.Headers {
+		if headerValue.Key == ":status" {
+			code, _ := strconv.Atoi(string(headerValue.RawValue))
+			if code != 200 {
```

Comment:
30x redirection are successful codes. Seems we will never have such case? Is it possible to have 20x?  here? better to have a helper method and let's consider those cases

Reply:
if we have the helper, we can further simplify the return results, just return the code is enough.

Reply:
I checked the code, codes returned are 200, 400, 401, 404 and 500. HandleResponseBody should only be called if response has ChatCompletitions object without that it will throw error.